backend/app/adapter/gqlapi/resolver/authmutation_test.go

```diff
@@ -0,0 +1,235 @@
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				entity.User{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: newLongLink,
+				},
+			},
+			shouldError: false,
+		},
+		{
+			name: "update both alias and long link",
+			args: &UpdateURLArgs{
+				OldAlias: "mySimpleAlias",
+				URL: URLInput{
+					CustomAlias: &newAlias,
+					OriginalURL: &newLongLink,
+				},
+			},
+			user: entity.User{
+				ID:    "1",
+				Email: "short@gmail.com",
+			},
+			urls: urls,
+			relationUsers: []entity.User{
+				entity.User{
+					ID:    "1",
+					Email: "short@gmail.com",
+				},
+			},
+			relationShortLinks: []entity.ShortLink{
+				entity.ShortLink{
+					Alias:    "mySimpleAlias",
+					LongLink: "https://www.google.com/",
+				},
+			},
+			expectedShortLink: &URL{
+				url: entity.ShortLink{
+					Alias:    newAlias,
+					LongLink: newLongLink,
+				},
+			},
+			shouldError: false,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			blockedHash := map[string]bool{
+				"http://malware.wicar.org/data/ms14_064_ole_not_xp.html": false,
+			}
+			blacklist := risk.NewBlackListFake(blockedHash)
+			shortLinkRepo := repository.NewShortLinkFake(testCase.urls)
+			userShortLinkRepo := repository.NewUserShortLinkRepoFake(
+				testCase.relationUsers,
+				testCase.relationShortLinks,
+			)
+
+			keyFetcher := keygen.NewKeyFetcherFake([]keygen.Key{})
+			keyGen, err := keygen.NewKeyGenerator(2, &keyFetcher)
+			assert.Equal(t, nil, err)
+
+			longLinkValidator := validator.NewLongLink()
+			aliasValidator := validator.NewCustomAlias()
+			riskDetector := risk.NewDetector(blacklist)
+
+			tm := timer.NewStub(now)
+			changeLogRepo := repository.NewChangeLogFake([]entity.Change{})
+			userChangeLogRepo := repository.NewUserChangeLogFake(map[string]time.Time{})
+			changeLog := changelog.NewPersist(keyGen, tm, &changeLogRepo, &userChangeLogRepo)
+
+			tokenizer := crypto.NewTokenizerFake()
+			auth := authenticator.NewAuthenticator(tokenizer, tm, time.Hour)
+
+			authToken, err := auth.GenerateToken(testCase.user)
+			assert.Equal(t, nil, err)
+
+			creator := url.NewCreatorPersist(
+				&shortLinkRepo,
+				&userShortLinkRepo,
+				keyGen,
+				longLinkValidator,
+				aliasValidator,
+				tm,
+				riskDetector,
+			)
+			updater := url.NewUpdaterPersist(
+				&shortLinkRepo,
+				&userShortLinkRepo,
+				keyGen,
+				longLinkValidator,
+				aliasValidator,
+				tm,
+				riskDetector,
+			)
+			authMutation := newAuthMutation(
+				&authToken,
+				auth,
+				changeLog,
+				creator,
+				updater,
+			)
+			shortLink, err := authMutation.UpdateURL(testCase.args)
+			assert.Equal(t, nil, err)
```

Comment:
```suggestion
if testCase.shouldError {
			assert.NotEqual(t, nil, err)
			return
}
assert.Equal(t, nil, err)
```
I remember violating assertion won't terminate the test. I guess this is because we didn't force fail fast mode.